Add SSReflect contextual pattern `UNDER` (#19011)
rocq-prover/rocq: doc/changelog/07-ssreflect/19011-add-under-pattern.rst

- **Added:**

>   A new SSReflect contextual pattern ``UNDER`` is provided to restrict the scope
>   of :tacn:`rewrite <rewrite (ssreflect)>` to the current :tacn:`under` context.
>   This pattern is useful to workaround the issue that occurs when we rewrite an
>   equality directly involving the main bound variable.
>   Now we can write: :g:`rewrite [UNDER]lem.`, or :g:`rewrite [in UNDER]lem.`

Reviewer: ```suggestion
  Contextual pattern ``UNDER``, which restricts the scope
  of :tacn:`rewrite <rewrite (ssreflect)>` to the current :tacn:`under` context.
  This pattern is useful to work around the issue that occurs when rewriting an
  equality directly involving the main bound variable.
  Now you can write: :g:`rewrite [UNDER]lem.`, or :g:`rewrite [in UNDER]lem.`
```

  (`#19011 <https://github.com/coq/coq/pull/19011>`_,

>   fixes `#11118 <https://github.com/coq/coq/issues/11118>`_,

Reviewer: ```suggestion
  workaround for`#11118 <https://github.com/coq/coq/issues/11118>`_,
```

IIUC this doesn't fix the underlying bug, right?

Author: Actually I wouldn't say #11118 is a bug of the tactic `under`, because the issue just comes from a "too-wide rewrite" (manually written by the Coq user) and the only tricky thing is that it only fails at the `over` time.

So this PR is not just a "fix" but a "documented solution" to manually fix one's rewrite tactic, along with a "convenience shortcut" for this solution.

Also regarding your question about fix vs. workaround, there is the question whether we should close #11118 or not upon the merge of #19011.
IMO, I think so.

Author: @gares @proux01 what do you think some opinion on that point?

Author: @jfehrle on second thought, maybe we can just:

* write `workaround for #11118` in the changelog as you suggested,
* and still close #11118 in the end.

I'll squash and force-push this small change, sorry for the noise…
